File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.cs
public class Solution {
    public int solution(string s) {
        int answer = 0;
        int count = 0;
        string now = "";
        
        foreach (char c in s)
        {
            now += c;
            
            if (c == now[0])
                count += 1;
            
            else if (count != 0 && now.Length >= 2 * count)
            {
                count = 0;
                now = "";
                answer += 1;
            }
        }
        
        return (now.Length > 0)? answer + 1 : answer;
    }
}

File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    count = 0
    now = ''
    
    for c in s:
        now += c

        if c == now[0]:
            count += 1

        elif count != 0 and len(now) >= count * 2:
            now = ''
            count = 0
            answer += 1
            
    return answer + int(now != '')